posts/Jan/2022-04-29-mapping-basic-custom-type-in-hibernate-orm-600.adoc

```diff
@@ -0,0 +1,319 @@
+= Custom basic type mapping in Hibernate 6
+:icons: font
+Jan Schatteman <jschatte@redhat.com>
+
+This is a blog post about the basic type mapping system in Hibernate 6, which has changed quite a bit with respect to previous versions, specifically in how individual parts that make up a basic type mapping are configured. Before 6.0, the user could implement and register a custom org.hibernate.type.BasicType contract, but this is now considered deprecated.
```

Comment:
```suggestion
Much has changed in ORM 6.0 with regard to its type system.  

The https://docs.jboss.org/hibernate/orm/6.0/userguide/html_single/Hibernate_User_Guide.html#basic[User Guide]
does a good job covering the mapping of basic values.  Migrating users will also find the
https://docs.jboss.org/hibernate/orm/6.0/migration-guide/migration-guide.html[Migration Guide] helpful.  This post
digs a little deeper into the effects of those changes on the handling and mapping of basic values.  


Historically Hibernate provided 3 ways to customize the mapping of a basic value -

* `UserType`
* `BasicType`
* `AttributeConverter`


6.0 removes the ability to customize mappings through a custom `BasicType`, but adds a new "compositional" approach.


// from here, I'd break it down into sections :

== UserType

// mostly the same - only reading method changed (name to position)


== AttributeConverter

// completely the same


== Compositional

// here is the meat
This is a blog post about the basic type mapping system in Hibernate 6, which has changed quite a bit with respect to previous versions, specifically in how individual parts that make up a basic type mapping are configured. Before 6.0, the user could implement and register a custom org.hibernate.type.BasicType contract, but this is now considered deprecated.
```

Reply:
Not sure why it kept the "This is a blog post about the basic type mapping system in Hibernate 6, ..." part in my suggestion :D

Reply:
ehhh, probably because you left it in there, either that or GH already has AI sniffing for outstanding prose in place :rofl: